app/models/infrastructure.rb

```diff
@@ -52,10 +52,14 @@ def self.setup(params)
   end
 
   def add_component(attrs, seq)
+    category = attrs[:type] || attrs['type']
```

Comment:
we should rename `category` to `component_type` in `InfrastructureComponent
the naming is a bit confusing

Reply:
Done
https://github.com/BaritoLog/BaritoMarket/commit/09a9231fc6d0860ec37267b24f1970102f74f2d9
https://github.com/BaritoLog/BaritoMarket/commit/f7cef9606053c57e251ecd33cd0bd0780d628edb
https://github.com/BaritoLog/BaritoMarket/commit/40b491de846206c1aed2b7096faf97b124639234
https://github.com/BaritoLog/BaritoMarket/commit/124bd33e6bee7678c1fd595ae51bfb1ecaadcb2e